src/crab/ebpf_domain.cpp

```diff
@@ -119,243 +121,674 @@ static std::vector<linear_constraint_t> assume_bit_cst_interval(const NumAbsDoma
+
+    if (right_interval <= interval_t::nonnegative_int(false)) {
+        // Interval can be represented as both an svalue and a uvalue since it fits in [0, INT_MAX].
+        auto lpub = left_interval_positive.truncate_to_sint(false).ub();
+        return {left_svalue >= 0,
+                strict ? left_svalue > right_svalue : left_svalue >= right_svalue,
+                left_svalue <= left_uvalue,
+                left_svalue >= left_uvalue,
+                left_uvalue >= 0,
+                strict ? left_uvalue > right_uvalue : left_uvalue >= right_uvalue,
+                left_uvalue <= *lpub.number()};
+    } else if ((left_interval_negative | left_interval_positive) <= interval_t::negative_int(false) &&
+               right_interval <= interval_t::negative_int(false)) {
+        // Interval can be represented as both an svalue and a uvalue since it fits in [INT_MIN, -1],
+        // aka [INT_MAX+1, UINT_MAX].
+        return {left_uvalue >= (number_t{INT32_MAX} + 1),
+                strict ? (left_uvalue > right_uvalue) : (left_uvalue >= right_uvalue),
+                strict ? (left_svalue > right_svalue) : (left_svalue >= right_svalue)};
+    } else if (inv.eval_interval(left_svalue) <= interval_t::signed_int(false) &&
+               inv.eval_interval(right_svalue) <= interval_t::signed_int(false)) {
+        // Interval can only be represented as an svalue.
+        return {strict ? (left_svalue > right_svalue) : (left_svalue >= right_svalue)};
+    } else {
+        // We can't directly compare the svalues since they may differ in high order bits.
+        return {};
+    }
+}
+
+static std::vector<linear_constraint_t> assume_signed_cst_interval(const NumAbsDomain& inv, Condition::Op op, bool is64,
+                                                                   variable_t left_svalue, variable_t left_uvalue,
+                                                                   const linear_expression_t& right_svalue,
+                                                                   const linear_expression_t& right_uvalue) {
+    using crab::interval_t;
+    using namespace crab::dsl_syntax;
+
+    interval_t left_interval = interval_t::bottom();
+    interval_t right_interval = interval_t::bottom();
+    interval_t left_interval_positive = interval_t::bottom();
+    interval_t left_interval_negative = interval_t::bottom();
+    get_signed_intervals(inv, is64, left_svalue, left_uvalue,
+                         right_svalue, left_interval,
+                         right_interval, left_interval_positive,
+                         left_interval_negative);
+
+    if (op == Condition::Op::EQ) {
+        // Handle svalue == right.
+        if (is64)
+            return assume_signed_64bit_eq(inv, left_svalue, left_uvalue, right_interval, right_svalue, right_uvalue);
+        else
+            return assume_signed_32bit_eq(inv, left_svalue, left_uvalue, right_interval);
+    }
```

Comment:
Not too important, but now this can easily be extracted out of this function, and there is no need for get_signed_intervals() for it any more, since it only needs the right interval. Maybe that's just a TODO.